routines/populate.go

```diff
@@ -0,0 +1,50 @@
+package routines
+
+import (
+	"log"
+	"time"
+
+	"github.com/csivitu/bl0b/ctftime"
+	"github.com/csivitu/bl0b/db"
+)
+
+func populate(t time.Time) {
+	log.Println("Populating database with upcoming CTFs!")
+	DB := db.New()
+	defer DB.Close()
+
+	ctf := ctftime.New()
+
+	now := t.Unix()
+	events, err := ctf.GetEvents(10, now, now+60*60*24*7)
+
+	if err != nil {
+		log.Println(err)
+	}
+
+	err = DB.AddEvents(&events)
+	if err != nil {
+		log.Println(err)
+	}
+}
+
+// Populate run every interval and adds items to the database
+func Populate(t time.Duration) {
+	ticker := time.NewTicker(t)
+
+	done := make(chan bool)
+
+	go func() {
```

Comment:
Right, but `func setInterval(f func(), t time.Duration) chan bool {` will only be able to take a function which has no return type and no parameters. Is there a way to make it take any function?

I can think of 
```go
func setInterval(f func(interface{}) interface{}, t time.Duration) chan bool {
...
}
```
But again, that would be able to take just 1 parameter of any type I think.

Reply:
You could do that, and there is a way to make it any number of params using the ... syntax too, but i don't recommend that

I meant it to be used with an anonymous function wrapper, as in the example I gave, that is much cleaner and doesn't do any type checking hacks

Reply:
Instead of an anonymouse function `func()`, I passed an anonymous function `func(t time.Time)`, so that you can pass the time from `ticker.C` to the function, then it can decide whether or not to use it.

```go
func SetInterval(f func(time.Time), t time.Duration) chan bool {
	ticker := time.NewTicker(t)

	done := make(chan bool, 1)

	go func() {
		defer ticker.Stop()
		for {
			select {
			case t := <-ticker.C:
				f(t)
			case <-done:
				return
			}
		}
	}()
	return done
}
```

A function which does not want `ticker.C` can be wrapped like this:

```go
SetInterval(
    func(_ time.Time) {
        otherFunc()
    },
    time
}
```

@thebongy verify in the commit once though.

Reply:
I don't think it makes sense to pass the timer in

And the populate function can just use time.now(), i don't see why it would do anything else

Reply:
In general, in a utility, you might need the time when the tick ran though. Of course, in this case, it can use `time.Now()` but I think it's a good idea to pass the tick time.